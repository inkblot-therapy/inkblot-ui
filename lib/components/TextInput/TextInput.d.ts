import * as React from 'react';
export interface TextInputProps {
    /** label */
    label: string;
    /** name */
    name: string;
    /** placeholder */
    placeholder?: string;
    /** label style */
    labelStyle?: object;
    /** input style */
    inputStyle?: object;
    /** disabled */
    disabled?: boolean;
    /** error */
    error?: boolean;
    /** type */
    type?: string;
    /** className of surrounding div .className > label, .className >input */
    className?: string;
    /** inline message, good for error text */
    inlineMessage?: string;
    /** default value of the input */
    value?: string;
    /** onChange  */
    onChange?: (event: object) => void;
}
export default class TextInput extends React.Component<TextInputProps> {
    static defaultProps: TextInputProps;
    render(): JSX.Element;
}
