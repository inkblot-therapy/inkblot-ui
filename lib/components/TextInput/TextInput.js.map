{"version":3,"file":"TextInput.js","sourceRoot":"","sources":["../../../src/components/TextInput/TextInput.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,6BAA+B;AAC/B,4DAAuD;AA6BvD;IAAuC,6BAA+B;IAAtE;;IAWA,CAAC;IAHQ,0BAAM,GAAb;QACE,OAAO,oBAAC,yBAAe,eAAK,IAAI,CAAC,KAAK,EAAI,CAAC;IAC7C,CAAC;IATa,sBAAY,GAAmB;QAC3C,QAAQ,EAAE,KAAK;QACf,KAAK,EAAE,OAAO;QACd,IAAI,EAAE,EAAE;QACR,IAAI,EAAE,MAAM;KACb,CAAC;IAKJ,gBAAC;CAAA,AAXD,CAAuC,KAAK,CAAC,SAAS,GAWrD;kBAXoB,SAAS"}