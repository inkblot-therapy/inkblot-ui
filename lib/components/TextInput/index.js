"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TextInput_1 = require("./TextInput");
exports.default = TextInput_1.default;
//# sourceMappingURL=index.js.map