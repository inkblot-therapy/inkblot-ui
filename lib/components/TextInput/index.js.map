{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/components/TextInput/index.tsx"],"names":[],"mappings":";;AAAA,yCAAsC;AAA7B,8BAAA,OAAO,CAAA"}