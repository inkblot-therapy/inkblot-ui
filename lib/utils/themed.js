"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var renderer = require("react-test-renderer");
var styled_components_1 = require("styled-components");
var theme_1 = require("../theme");
var themed = function (component) { return renderer.create(React.createElement(styled_components_1.ThemeProvider, { theme: theme_1.default }, component)); };
exports.default = themed;
//# sourceMappingURL=themed.js.map