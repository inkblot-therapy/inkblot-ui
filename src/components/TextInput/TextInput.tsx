import * as React from 'react';
import StyledTextInput from './styled/StyledTextInput';

export interface TextInputProps {
  label: string;
  placeholder?: string;
  style?: {};
  disabled?: boolean;
  error?: boolean;
  className?: string;
  value?: string;
  onChange?: (v?: {}) => void;
}

export default class TextInput extends React.Component<TextInputProps> {
  public static defaultProps: TextInputProps = {
    disabled: false,
    label: 'Label',
    onChange: undefined,
  };

  public render() {
    return <StyledTextInput {...this.props} />;
  }
}
