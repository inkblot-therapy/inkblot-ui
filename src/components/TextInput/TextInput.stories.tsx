import { boolean, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { wInfo } from '../../utils/wInfo';
import TextInput from './TextInput';

const stories = storiesOf('Components/TextInput', module);
stories.addDecorator(withKnobs);

stories.add(
  'default',
  wInfo()(() => (
    <TextInput
      error={boolean('error', false)}
      className={text('className')}
      name={text('name')}
      disabled={boolean('disabled', false)}
      label={text('Label', 'Basic')}
      placeholder={text('placeholder', 'Placeholder')}
      type={text('type', 'text')}
      inlineMessage={text('Inline Message', '')}
    />
  )),
);

stories.add(
  'optional styles',
  wInfo()(() => (
    <TextInput
      label={text('Label', 'I got Styles')}
      placeholder={text('value', 'Placeholder')}
      labelStyle={{ color: 'red' }}
      inputStyle={{ backgroundColor: '#dab420' }}
    />
  )),
);

const onChange = (v: object) => {
  console.log(
    `%c ${v.target.value} `,
    'font-weight: bold; font-size: 50px; color: #dab420;',
  );
};

stories.add(
  'optional callback ',
  wInfo()(() => (
    <TextInput label={text('Label', 'Check Console')} onChange={onChange} />
  )),
);

stories.add(
  'inline message',
  wInfo()(() => (
    <TextInput
      error={boolean('error', false)}
      label={text('Label', 'Inline Message')}
      placeholder={text('value', 'Placeholder')}
      inlineMessage={text('Inline Message', 'This is an Inline Message')}
    />
  )),
);
