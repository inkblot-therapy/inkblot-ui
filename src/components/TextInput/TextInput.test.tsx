import * as React from 'react';

import themed from '../../utils/themed';
import TextInput from './TextInput';

describe('TextInput', () => {
  it('should render correctly', () => {
    const component = themed(<TextInput>Test</TextInput>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
