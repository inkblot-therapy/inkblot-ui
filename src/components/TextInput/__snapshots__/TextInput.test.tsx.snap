// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
.c1 {
  min-width: 200px;
  height: 40px;
  padding-left: 10px;
  padding-right: 10px;
  border-radius: 4px;
  font-family: "Source Sans Pro",sans-serif;
  font-size: 16px;
  color: "#0f2045";
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  border: solid transparent;
  background-color: #fafafa;
  -webkit-transition: 0.25s;
  transition: 0.25s;
  cursor: auto;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 10px 0 rgba(99,140,177,0.2);
  background-color: #ffffff;
}

.c1::-webkit-input-placeholder {
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: rgba(15,32,69,0.75);
}

.c1::-moz-placeholder {
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: rgba(15,32,69,0.75);
}

.c1:-ms-input-placeholder {
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: rgba(15,32,69,0.75);
}

.c1::placeholder {
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: rgba(15,32,69,0.75);
}

.c0 {
  color: red;
  font-family: "Source Sans Pro",sans-serif;
  font-size: 16px;
  font-weight: 600;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #0f2045;
  display: block;
  margin-bottom: 20px;
}

.c2 {
  font-family: "Source Sans Pro",sans-serif;
  margin-top: 10px;
  font-size: 12px;
  font-weight: 600;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  -webkit-transition: 0.25s;
  transition: 0.25s;
  color: rgba(15,32,69,0.75);
}

<div>
  <label
    className="c0"
  >
    Label
  </label>
  <input
    className="c1"
    disabled={false}
    name=""
    onChange={[Function]}
    type="text"
  />
  <p
    className="c2"
  />
</div>
`;
